base/util.jl

```diff
@@ -134,8 +112,13 @@ function padded_nonzero_print(value,str)
 end
 
 function time_print(elapsedtime, bytes, gctime, allocs)
-    mt, pptime, fraction = prettyprint_nanoseconds(elapsedtime)
-    (fraction != 0) ? @printf("%4d.%03d %s", pptime, fraction, _sec_units[mt]) : @printf("%8d %s",  pptime, _sec_units[mt])
+    seconds = elapsedtime/1e9
+    if 1 <= round(seconds,3)
+        @printf("%.3f seconds", seconds)
+    else
+        power = -floor(log10(seconds))
+        @printf("%.3f/10^%d seconds", seconds*10^power, power)
```

Comment:
What about engineering notation?
Scientific notation seems a lot harder to read when comparing similar numbers,
you always have to be looking at 2 numbers, instead of just one in most cases,
and shifting them in your head before you can compare them.